.github/actions/cargo_target_dir_cache/action.yml

```
@@ -3,10 +3,12 @@ description: CARGO_TARGET_DIR Cache
 runs:
   using: "composite"
   steps:
+    - run: rustc --version > rustc_version.txt && cat rustc_version.txt
+      shell: bash
     - uses: actions/cache@937d24475381cd9c75ae6db12cb4e79714b926ed
       with:
         path: target
-        key: cache-cargo-target-${{ github.workflow }}-${{ github.job }}-ghcr.io/build-trust/ockam-builder@sha256:e43dd94652096b03cc472a3c709c7335e8b166cab77b7a7b56f88fa38f3d24cc-${{ hashFiles('**/Cargo.lock') }}
+        key: cache-cargo-target-${{ github.workflow }}-${{ github.job }}-ghcr.io/build-trust/ockam-builder@sha256:e43dd94652096b03cc472a3c709c7335e8b166cab77b7a7b56f88fa38f3d24cc-${{ hashFiles('**/Cargo.lock') }}-${{ hashFiles('./rustc_version.txt') }}
         restore-keys: cache-cargo-target-${{ github.workflow }}-${{ github.job }}-ghcr.io/build-trust/ockam-builder@sha256:e43dd94652096b03cc472a3c709c7335e8b166cab77b7a7b56f88fa38f3d24cc-
```

```suggestion
        key: cache-cargo-target-${{ hashFiles('./rustc_version.txt') }}-${{ github.workflow }}-${{ github.job }}-ghcr.io/build-trust/ockam-builder@sha256:e43dd94652096b03cc472a3c709c7335e8b166cab77b7a7b56f88fa38f3d24cc-${{ hashFiles('**/Cargo.lock') }}
        restore-keys: cache-cargo-target-${{ hashFiles('./rustc_version.txt') }}-${{ github.workflow }}-${{ github.job }}-ghcr.io/build-trust/ockam-builder@sha256:e43dd94652096b03cc472a3c709c7335e8b166cab77b7a7b56f88fa38f3d24cc-
```

We need to ensure that rust-nightly restores caches only for nightly cache uploads

---

As I see it, the 'rustc version' dependency can be treated just like the 'Cargo.lock files' dependency. And 'Cargo.lock files' is not in `restore-keys` (which I think is correct).

> We need to ensure that rust-nightly restores caches only for updated nightly version cache uploads

Maybe I don't understand. 😕 Are you concerned, for example, that the `test` and `test_nightly` caches are the same? As I see it, they are currently kept separate by the `${{ github.job }}` part of the key. 

Thanks for you patience with me!

---

> Maybe I don't understand. 😕 Are you concerned, for example, that the test and test_nightly caches are the same? As I see it, they are currently kept separate by the ${{ github.job }} part of the key.

Hmmm, I see. This is entirely correct. Thanks. Resolving conversation now.